proj/src/main/webapp/chat.js

```diff
@@ -918,7 +931,20 @@ function changeChatOnClick(domElement, tag, chatId) {
         sessionStorage.clear(); // clear pfps and activeChat data
         sessionStorage.activeChatTag = tag;
         sessionStorage.activeChatId = chatId;
-        initRef();
+        
+        var lastVisited;
+        var uid = firebase.auth().currentUser.uid;
+        const dbRef = firebase.database().ref("/users/" + uid + "/lastVisited/");
+        dbRef.once('value', snap => {
+            lastVisited = snap.child(tag).val();
+            initRef(lastVisited);
+        }).finally(function(){
+            var date = new Date();
+            var timestamp = date.getTime();
+            var updates = {};
+            updates[tag] = timestamp;
+            dbRef.update(updates);
```

Comment:
Should this be updating the timestamp for the current tag or the tag we are leaving or neither?

Reply:
It's currently updates the lastVisited property for the tag that's clicked on after the dom updates